FungibleToken.ts

```diff
@@ -123,14 +116,48 @@ export class FungibleToken extends TokenContract implements FungibleTokenLike {
     return balance
   }
 
-  @method.returns(UInt64)
-  async getSupply() {
-    return this.supply.getAndRequireEquals()
+  /** This function is used to fold the actions from minting/burning */
+  private calculateCirculating(
+    oldCirculating: UInt64,
+    pendingActions: MerkleList<MerkleList<Int64>>,
+  ): UInt64 {
+    let newCirculating: Int64 = this.reducer.reduce(
+      pendingActions,
+      Int64,
+      (circulating: Int64, action: Int64) => {
+        return circulating.add(action)
+      },
+      Int64.from(oldCirculating),
+      { maxUpdatesWithActions: 10 },
```

Comment:
I think we can be more aggressive here. How about `maxUpdatesWithActions: 100` or `500`? To reduce the likelihood that there are too many actions to process